### PR TITLE
feat: add bit number of the data elements

### PR DESCRIPTION
**Pull request by lucaslima-cw**
Closes cloudwalk/product-clearing#1398

### CODE REVIEW

File: src/iso_field.rs [hunk 1]
```diff
@@ -6,12 +6,14 @@
...
 
+use serde::Serializer;
+
 #[derive(Debug, PartialEq, Serialize, Deserialize, Clone)]
 pub enum FieldCharType {
     Iso8583_n,
     Iso8583_ns,
     Iso8583_xn,
-    ISO8583_a,
+    Iso8583_a,
```
Comment (renancloudwalk) [inline src/iso_field.rs:16]: Does this type exist at all? 🤔

File: src/iso_field.rs [hunk 2]
```diff
@@ -103,6 +105,7 @@ impl FieldSizeType {
 #[derive(Debug, PartialEq, Serialize, Deserialize)]
 pub struct IsoField {
     pub label: String,
+    pub de: String,
```
Comment (renancloudwalk) [inline src/iso_field.rs:108]: Please call this `data_element` we are not short on characters here 😆

**Review by renancloudwalk — CHANGES_REQUESTED** [review]

File: src/lib.rs [hunk 3]
```diff
@@ -13,114 +13,90 @@ pub mod iso_specs;
...
-impl Group {
+impl Message {
     /// Returns a HashMap [message.label => message.utf8_value] of all self.messages
     pub fn get_messages_hash(&self) -> Result<HashMap<String, String>> {
         let messages_hash: HashMap<String, String> = self
-            .messages
-            .iter()
-            .map(|p| (p.get_label(), p.utf8_value()))
+            .data_elements
+            .clone()
```
Comment (bizzoni) [inline src/lib.rs:64]: U really need this` .clone()` here?

Comment (lucaslima-cw) [reply src/lib.rs:64]: I'll check that. Thanks!

Comment (lucaslima-cw) [reply src/lib.rs:64]: image

File: src/lib.rs [hunk 4]
```diff
@@ -13,114 +13,90 @@ pub mod iso_specs;
...
+    pub mti: String,
     pub category: Category,
+    pub primary_bitmap: [u8; 8],
+    pub data_elements: HashMap<String, iso_field::IPMValue>,
+    pub pds: HashMap<String, String>,
 }
 
-impl Group {
+impl Message {
     /// Returns a HashMap [message.label => message.utf8_value] of all self.messages
```
Comment (lucaslima-cw) [inline src/lib.rs:60]:
```suggestion
    /// Returns a HashMap [message.label_id => message.utf8_value] of all self.messages
```

File: src/lib.rs (hunk 3 above)
Comment (bizzoni) [reply src/lib.rs:64]: `        let messages_hash: HashMap<String, String> = self
            .data_elements
            .iter()
            .map(|(k, v)| (k.clone(), v.get_string()))
            .collect();`

File: tests/integration_test.rs [hunk 5]
```diff
@@ -104,21 +105,53 @@ fn parse_t113_deblocked_sample() {
...
         }
         println!("\n\n{:?}\n\n", &original_group.pds);
 
         println!("\n\n");
-        let group = iso8583_file.clone().groups.get(*g).unwrap().clone();
-        let messages = group.messages.clone();
-        for m in messages {
-            println!("post: {} => {}", &m.get_label(), &m.utf8_value());
+        let message = iso8583_file.clone().messages.get(*g).unwrap().clone();
+        let messages = message.data_elements.clone();
```
Comment (Vini97CLeite) [inline tests/integration_test.rs:125]: Please, change to data_elements

File: src/lib.rs [hunk 6]
```diff
@@ -13,114 +13,90 @@ pub mod iso_specs;
...
+    pub primary_bitmap: [u8; 8],
+    pub data_elements: HashMap<String, iso_field::IPMValue>,
+    pub pds: HashMap<String, String>,
 }
 
-impl Group {
-    /// Returns a HashMap [message.label => message.utf8_value] of all self.messages
+impl Message {
+    /// Returns a HashMap [message.label_id => message.utf8_value] of all self.messages
     pub fn get_messages_hash(&self) -> Result<HashMap<String, String>> {
```
Comment (Vini97CLeite) [inline src/lib.rs:61]: Considering that is a legacy code, let's remove it.

File: src/lib.rs [hunk 7]
```diff
@@ -155,17 +131,17 @@ impl Iso8583File {
...
-            let group_messages_hash = group.get_messages_hash().expect("Unable to find message hash for group. Maybe the file is broken");
+        // since this loops inside each message and uses a hash to find a match, this is memory intensive
+        for message in self.messages {
+            let message_hash = message.get_messages_hash().expect("Unable to find message hash for message. Maybe the file is broken");
             for search_key in search.keys() {
-                match group_messages_hash.get(search_key) {
-                    Some(message) => {
-                        if search.get(search_key).unwrap().contains(message) {
-                            search_groups_result.push(group);
+                match message_hash.get(search_key) {
```
Comment (Vini97CLeite) [inline src/lib.rs:141]: Since, we'll remove some legacy code we'll have to adapt this to consume the new structs.

**Review by Vini97CLeite — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 6 above)
Comment (bizzoni) [reply src/lib.rs]: `get_messages_hash`  is a method I created to use here at Receivables, we need this method. Why do you guys wanna remove it? Always remember that we use this lib too.

File: src/lib.rs [hunk 8]
```diff
@@ -13,114 +13,79 @@ pub mod iso_specs;
...
+    pub mti: String,
     pub category: Category,
+    pub primary_bitmap: [u8; 8],
+    pub data_elements: HashMap<String, iso_field::IPMValue>,
+    pub pds: HashMap<String, String>,
 }
 
-impl Group {
-    /// Returns a HashMap [message.label => message.utf8_value] of all self.messages
-    pub fn get_messages_hash(&self) -> Result<HashMap<String, String>> {
```
Comment (bizzoni) [inline src/lib.rs:72]: Please keep it

**Review by bizzoni — CHANGES_REQUESTED** [review]

**Review by bizzoni — APPROVED** [review]

**Review by Vini97CLeite — APPROVED** [review]

**Pull request merged**